.github/workflows/main.yml

```diff
@@ -26,54 +30,137 @@ jobs:
+            echo "ERROR: NIX_CACHE_KEY_PUB hardcoded value does not match" \
+            "the reference stored in SSM ('$param'). The SSM value should be considered" \
+            "authoritative. Please update the hardcoded value in the workflow file.";
+            exit 1; \
+          }
+
+      - name: Get nix_cache_key_sec from SSM
+        if: steps.calculate_write_flag.outputs.has_write
+        id: get_nix_cache_key_sec
+        run: |
+          param=$(aws ssm get-parameter \
+            --name arn:aws:ssm:${{ env.AWS_REGION }}:${{ env.AWS_ACCOUNT_ID }}:parameter${{ env.SSM_CONFIG_PATH }}/nix_cache_key_sec \
+            --with-decryption \
+            | jq -r '.Parameter.Value')
+          echo "$param" | sudo cp /dev/stdin /private/nix-signing-key
+
+      - name: Create /etc/nix/upload-to-cache.sh
+        if: steps.calculate_write_flag.outputs.has_write
+        run: |
+          sudo mkdir -p /etc/nix
+          sudo tee /etc/nix/upload-to-cache.sh <<EOL
+          #!/bin/bash
+
+          set -f # disable globbing
+          export IFS=' '
+          echo "Signing and uploading paths" \$OUT_PATHS
+
+          exec /nix/var/nix/profiles/default/bin/nix copy --to '${{ env.S3_LOCATION_NIX_CACHE }}?region=${{ env.AWS_REGION }}&secret-key=/private/nix-signing-key' \$OUT_PATHS
+          EOL
+          sudo chmod a+x /etc/nix/upload-to-cache.sh
+
+      - name: Expose AWS credentials to the nix-daemon
+        if: steps.calculate_write_flag.outputs.has_write
+        run: |
+          sudo launchctl setenv AWS_ACCESS_KEY_ID     "$AWS_ACCESS_KEY_ID"
+          sudo launchctl setenv AWS_SECRET_ACCESS_KEY "$AWS_SECRET_ACCESS_KEY"
+          sudo launchctl setenv AWS_SESSION_TOKEN     "$AWS_SESSION_TOKEN"
+
+      # https://github.com/NixOS/nix/issues/2242#issuecomment-2336841344
+      - name: macOS 15 eDSRecordAlreadyExists workaround
+        run: echo "NIX_FIRST_BUILD_UID=30001" >> "$GITHUB_ENV"
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@v27
+        with:
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+          extra_nix_config: |
+            max-jobs = 1
+            extra-substituters = ${{ env.S3_LOCATION_NIX_CACHE }}?region=${{ env.AWS_REGION }}
+            extra-trusted-public-keys = ${{ env.NIX_CACHE_KEY_PUB }}
+            ${{ steps.calculate_write_flag.outputs.has_write && 'post-build-hook = /etc/nix/upload-to-cache.sh' }}
+
+      - name: Show /etc/nix/nix.conf
+        run: cat /etc/nix/nix.conf
+
+      # The following "build" command can be used for debugging without having to wait a long time:
+      # nix-build -E 'let pkgs = import (builtins.fetchTarball "https://github.com/NixOS/nixpkgs/archive/nixos-24.05.tar.gz") {}; in pkgs.writeText "example" "001"'
+      # Counter 001 can be modified to check for cache hits or misses.
+      - name: Run check
+        run: nix build .#checks.${{ matrix.platform.nixArch }}.${{ matrix.check == 'nil' && 'others' || matrix.check }} -L
```

Comment:
Not sure we need to run nil tests of MacOS at all. Especially because we are on the way of deprecation of go CLI.

Reply:
What's wrong with running tests on a platform used by a fraction of the developers?

Reply:
It's not officially supported platform for nild as result it can slow down development process

Reply:
But after all, broken tests by some developers that CI missed can also slow down the development process.
In any case, we may change the decision to run tests on macOS later if there are real problems with it.